CalibFormats/HcalObjects/src/HcalText2DetIdConverter.cc

```diff
@@ -158,7 +166,24 @@ bool HcalText2DetIdConverter::init (const std::string& fFlavor, const std::strin
     mId = HcalDetId (sub, getField (1), getField (2), getField (3));
   }
   else if (flavorName == "HT") {
-    mId = HcalTrigTowerDetId (getField (1), getField (2));
+    // We use the depth to signal the "version" being used (RCT or 1x1 HF). RCT
+    // has a 0 in the 10s digit, whereas 1x1 has a 1. The ones digit is still
+    // used to indicate depth, although in the 1x1 case this must be 0, so we
+    // set it as such.
+    const int depth_field = getField(3);
+    const int ones = depth_field % 10;
+    const int tens = (depth_field - ones) / 10;
+    if (tens == 0) {
+      const int depth = ones;
+      const int version = 0;
+      mId = HcalTrigTowerDetId (getField (1), getField (2), depth, version);
+    } else if (tens == 1) {
+      const int depth = 0;
+      const int version = 1;
+      mId = HcalTrigTowerDetId(getField(1), getField(2), depth, version);
+    } else {
+      // Undefined version!
+    }
```

Comment:
As already pointed out by @slava77 in https://github.com/cms-sw/cmssw/pull/7950/files#r25869739 please avoid magic numbers (in this case `0` and `1` for the version, `10` for the digits.